packages/codemods/src/codemods/v0.39.x/updateRouterParamTypes/__testfixtures__/variableRef.input.tsx

```diff
@@ -0,0 +1,31 @@
+// In this file, all Page components from 'src/pages` are auto-imported. Nested
+// directories are supported, and should be uppercase. Each subdirectory will be
+// prepended onto the component name.
+//
+// Examples:
+//
+// 'src/pages/HomePage/HomePage.js'         -> HomePage
+// 'src/pages/Admin/BooksPage/BooksPage.js' -> AdminBooksPage
+
+import { Router, Route } from '@redwoodjs/router'
+
+const userRouteParamTypes = {
+  slug: {
+    constraint: /\w+-\w+/,
+    transform: (param) => param.split('-'),
+  },
+  something: {
+    constraint: /\w+.\w+/,
+    transform: (param) => param.split('.'),
+  },
+}
+
+const Routes = () => {
+  return (
+    <Router pageLoadingDelay={350} paramTypes={userRouteParamTypes}>
+      <Route notfound page={NotFoundPage} />
+    </Router>
+  )
+}
```

Comment:
If someone only has one custom param type, they might have done it like this. Do you support that as well?

```tsx
const slug = {
  constraint: /\w+-\w+/,
  transform: (param) => param.split('-'),
}

const Routes = () => {
  return (
    <Router pageLoadingDelay={350} paramTypes={{ slug }}>
      <Route notfound page={NotFoundPage} />
    </Router>
  )
}
```

Reply:
I don't think so. Will check.

Reply:
Added a couple of cases. Can you please check `partialEmbeds` test cases?